### PR TITLE
added charge estimation for losses

### PR DESCRIPTION
**Pull request by AlexTeslenko**
added charge estimation for losses

### CODE REVIEW

File: libs/ml/include/ml/charge_estimation/ops/constants.hpp [hunk 1]
```diff
@@ -33,6 +33,19 @@ static constexpr OperationsCount FLATTEN_PER_ELEMENT             = 1;
 static constexpr OperationsCount MULTIPLICATION_PER_ELEMENT      = 3;
 static constexpr OperationsCount PLACEHOLDER_READING_PER_ELEMENT = 0;
 static constexpr OperationsCount WEIGHTS_READING_PER_ELEMENT     = 0;
+static constexpr OperationsCount LOG_PER_ELEMENT                 = 1;
+static constexpr OperationsCount MAX_PER_ELEMENT                 = 1;
+static constexpr OperationsCount EXP_PER_ELEMENT                 = 1;
+static constexpr OperationsCount DIVISION_PER_ELEMENT            = 1;
+static constexpr OperationsCount CROSS_ENTROPY_PER_ELEMENT =
+    2 * SUBTRACTION_PER_ELEMENT + LOG_PER_ELEMENT;
+static constexpr OperationsCount MEAN_SQ_ERROR_PER_ELEMENT =
```
Comment (vhavrylov-ncube) [inline libs/ml/include/ml/charge_estimation/ops/constants.hpp:41]: I admire the idea of explaining what does a charge cost of a loss consist of using previously declared charges. Though I guess we need to double-check if these equations are correct.

**Review by vhavrylov-ncube — COMMENTED** [review]
LGTM but let other teammates approve too.

File: libs/ml/include/ml/charge_estimation/ops/constants.hpp (hunk 1 above)
Comment (baykaner) [reply libs/ml/include/ml/charge_estimation/ops/constants.hpp:41]: yes, but anyway all of this will need tweaking and adjustments in the next sprint along with benchmarking to verify the precision of the estimates

File: libs/ml/include/ml/charge_estimation/ops/constants.hpp [hunk 2]
```diff
@@ -33,6 +33,19 @@ static constexpr OperationsCount FLATTEN_PER_ELEMENT             = 1;
...
+    2 * MULTIPLICATION_PER_ELEMENT + ADDITION_PER_ELEMENT + SUBTRACTION_PER_ELEMENT;
+static constexpr OperationsCount SOFTMAX_PER_ELEMENT = MAX_PER_ELEMENT + SUBTRACTION_PER_ELEMENT +
+                                                       EXP_PER_ELEMENT + ADDITION_PER_ELEMENT +
+                                                       DIVISION_PER_ELEMENT;
+static constexpr OperationsCount CROSS_ENTROPY_SOFTMAX_PER_ELEMENT =
```
Comment (baykaner) [inline libs/ml/include/ml/charge_estimation/ops/constants.hpp]: traditionally this is called 'softmax cross entropy', not 'cross entropy softmax'.
Rename it please

**Review by baykaner — CHANGES_REQUESTED** [review]

**Review by baykaner — APPROVED** [review]
